#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/factory/containers/protos/WoodenBox.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/Floor.proto"

WorldInfo {
  basicTimeStep 5
}
Viewpoint {
  orientation 0.25864690457693157 0.16621936265967924 -0.951563398954469 2.043884757605146
  position 0.38686785236850896 0.7912596833189037 0.4794846427197156
  follow "wheel_bot"
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 5 5
}
DEF BOX_200 WoodenBox {
  translation -0.89336 -1.33336 0.1
  name "wooden box 200_0"
  size 0.2 0.2 0.2
  mass 1
}
DEF BOX_200 WoodenBox {
  translation 0.818645 -0.814603 0.1
  rotation 0 0 1 0.261799
  name "wooden box 200_1"
  size 0.2 0.2 0.2
  mass 1
}
DEF BOX_100 WoodenBox {
  translation -1.36464 -0.146134 0.05
  rotation 0 0 1 -0.523595307179586
  name "wooden box 100_0"
  size 0.1 0.1 0.1
  mass 1
}
DEF BOX_100 WoodenBox {
  translation 0.673451 1.56392 0.05
  rotation 0 0 -1 0.523595
  name "wooden box 100_1"
  size 0.1 0.1 0.1
  mass 1
}
DEF BOX_100 WoodenBox {
  translation 1.02293 -1.57076 0.05
  rotation 0 0 -1 0.523595
  name "wooden box 100_2"
  size 0.1 0.1 0.1
  mass 1
}
DEF BOX_200 WoodenBox {
  translation -0.75003 1.06734 0.1
  rotation 0 0 1 -0.261795307179586
  name "wooden box 200_2"
  size 0.2 0.2 0.2
  mass 1
}
DEF BOX_200 WoodenBox {
  translation 1.28768 0.606637 0.1
  rotation 0 0 1 0.523599
  name "wooden box 200_3"
  size 0.2 0.2 0.2
  mass 1
}
Solid {
  translation 0 2.505 0.05
  children [
    DEF WALL Shape {
      appearance PBRAppearance {
        baseColor 0.6 0.756863 0.945098
        roughness 0.5
        metalness 0
      }
      geometry Box {
        size 5 0.01 0.2
      }
    }
  ]
  name "wall0"
  boundingObject USE WALL
}
Solid {
  translation 0 -2.505 0.05
  children [
    USE WALL
  ]
  name "wall1"
  boundingObject USE WALL
}
Solid {
  translation 2.505 0 0.05
  rotation 0 0 1 1.5707963267948966
  children [
    USE WALL
  ]
  name "wall2"
  boundingObject USE WALL
}
Solid {
  translation -2.505 0 0.05
  rotation 0 0 1 1.5707963267948966
  children [
    USE WALL
  ]
  name "wall3"
  boundingObject USE WALL
}
Robot {
  translation 0 0 0.11
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 0.11 -0.05
      }
      device [
        RotationalMotor {
          name "motor_0"
        }
      ]
      endPoint DEF WHEEL0 Solid {
        translation -1.734723475976807e-18 0.10999999999999999 -0.05
        rotation 1 -7.850447875279516e-17 7.850447875279516e-17 1.5708
        children [
          DEF WHEEL_SHAPE Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              roughness 0.5
              metalness 0
              normalMap ImageTexture {
                url [
                  "../../red_brick_wall.jpg"
                ]
              }
            }
            geometry Cylinder {
              height 0.02
              radius 0.05
            }
          }
        ]
        name "wheel0"
        boundingObject USE WHEEL_SHAPE
        physics DEF WHEEL_PHY Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 -1 0
        anchor 0 -0.11 -0.05
      }
      device [
        RotationalMotor {
          name "motor_1"
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation 0 -0.1100000000000001 -0.05
        rotation 1 0 0 1.5708
        children [
          USE WHEEL_SHAPE
        ]
        name "wheel1"
        boundingObject USE WHEEL_SHAPE
        physics USE WHEEL_PHY
      }
    }
    DEF SHAPE_BODY Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.2 0.2 0.1
      }
    }
    Transform {
      translation 0.08 0 0.1
      scale 0.2 1 1
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 1
            roughness 1
            metalness 0
          }
          geometry Cone {
            bottomRadius 0.05
            height 0.1
          }
        }
      ]
    }
    InertialUnit {
      name "ins"
    }
    Accelerometer {
      name "accel"
    }
    Gyro {
    }
  ]
  name "wheel_bot"
  boundingObject USE SHAPE_BODY
  physics Physics {
  }
  controller "bench_robot_controller"
}
